### DISCUSSION TITLE
How to use under Ubuntu 18.04

### DISCUSSION THREAD

**Discussion by henfri** [ct-Open-Source/tuya-convert#1011] [General]
Hello,

just my notes, as the readme did not help me enough:

```
sudo bash
setxkbmap -layout de
apt-get update
add-apt-repository universe
service systemd-resolved stop

```
Then
```
git clone https://github.com/ct-Open-Source/tuya-convert
cd tuya-convert
./install_prereq.sh

```
Check the name of your wifi interface (eg wlp6s0)

`gedit config.txt` and enter your wifi interface under WLAN=

`./start_flash.sh
`
you will get an error in line 138 (terminated). You can ignore that.

If you know this, it is easy :-)

Thanks for tuya-convert!

**Comment by kueblc** [comment]
Thanks for sharing your experience @henfri!

Converting this to a discussion to keep it visible.

**Comment by Andreas2430** [comment]
Thank you very much @henfri ! 

I was following the guide in the [c't article ](https://www.heise.de/ct/artikel/Tuya-Convert-IoT-Geraete-ohne-Loeten-vom-Cloud-Zwang-befreien-4283623.html) and the readme, but both didn't mention that the universe repo needs to be added, so I was struggling because a lot of packages could not be installed. With your notes, everything worked fine. 

@kueblc it might be helpful for other users if this information could be added to the readme. Thanks for tuya-convert!

**Comment by henfri** [comment]
Great. Thanks for the feedback!

**Comment by isur3n** [comment]
Is it a must to change keyboard layout to german as mentioned by `setxkbmap -layout de` ?

**Reply by henfri** [reply]
No
